src/components/Feed/Feed.js

```diff
@@ -3,12 +3,38 @@ import { Route } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 import './Feed.css';
+import Post from '../../components/Post/Post';
+import { user, posts } from '../../Data/mockData';
 
-// import Post from '../../components/Post/Post';
+const Feed = ({ profileIsVisible }) => {
+  const [currentUserPosts, setCurrentUserPosts] = useState([])
+  const [allPosts, setAllPosts] = useState([])
+
+  useEffect(() => {
+    setCurrentUserPosts(user.posts)
+    setAllPosts(posts)
+  }, [currentUserPosts, allPosts])
+
+  const displayPostsForCurrentUser = (postsToDisplay) => {
+    return postsToDisplay.map(post => {
+      return (
+        <Post
+          userName={post.userName}
+          title={post.title}
+          date={post.date}
+          location={post.location}
+          description={post.description}
+          distance={post.distance}
+          imageUrl={post.image}
+        />
+      )
+    })
+  }
 
-const Feed = () => {
   return (
-    <h2>Feed goes here</h2>
+    <div className='post-container'>
+      {profileIsVisible ? displayPostsForCurrentUser(currentUserPosts) : displayPostsForCurrentUser(allPosts)}
+    </div>
```

Comment:
Nice! This answers the questions I had - looks good!

Reply:
I just realized the naming of that function is confusing. I updated it to `displayPostsInFeed` since it deals with two different feeds.